src/processor/processor.cpp

```diff
@@ -25,12 +25,16 @@ std::shared_ptr<FactorizedTable> QueryProcessor::execute(
     auto task = std::make_shared<ProcessorTask>(resultCollector, context);
     decomposePlanIntoTask(lastOperator->getChild(0), task.get(), context);
     initTask(task.get());
+    context->clientContext->progressBar->startProgress();
     taskScheduler->scheduleTaskAndWaitOrError(task, context);
     return resultCollector->getResultFactorizedTable();
 }
 
 void QueryProcessor::decomposePlanIntoTask(
     PhysicalOperator* op, Task* task, ExecutionContext* context) {
+    if (op->isSource()) {
+        context->clientContext->progressBar->addPipeline();
```

Comment:
This is irrelevant to this PR but I think it might be better to introduce Pipeline as another abstraction level.